docs/conf.py

```diff
@@ -262,6 +265,54 @@ def rstjinja(app, docname, source):
 open(target_file, 'a').close()
 
 
+class Generate(Directive):
+    """
+    Custom directive to include raw output generated using supplied Python code
+
+    The required argument specifies the format of the output (e.g., ``html``)
+
+    The optional flag ``html_border`` surrounds HTML output with a black border for visual
+    separation.
+    """
+    required_arguments = 1
+    optional_arguments = 0
+    option_spec = {'html_border': directives.flag}
+    has_content = True
+
+    def run(self):
+        # Respect the same disabling options as the ``raw`` directive
+        if (not self.state.document.settings.raw_enabled
+            or (not self.state.document.settings.file_insertion_enabled
+                and ('file' in self.options
+                     or 'url' in self.options))):
+            raise self.warning('"%s" directive disabled.' % self.name)
+
+        attributes = {'format': ' '.join(self.arguments[0].lower().split())}
+
+        source, lineno = self.state_machine.get_source_and_line(self.lineno)
+
+        old_stdout, sys.stdout = sys.stdout, StringIO()
+        try:
+            # Exceute the Python code and capture its output
+            exec('\n'.join(self.content))
+            text = sys.stdout.getvalue()
+
+            # Wrap HTML output in a black border if requested
+            if attributes['format'] == 'html' and 'html_border' in self.options:
+                text = f"<div style='border:1px solid black'>{text}</div>"
```

Comment:
Maybe adding a slightly larger gap between the border and the text might be nice?
```suggestion
                text = f"<div style='border:1px solid black; padding:3px'>{text}</div>"
```

Reply:
Good idea, done